src/libsync/propagatedownload.cpp

```diff
@@ -484,7 +486,22 @@ void PropagateDownloadFileQNAM::slotGetFinished()
         return;
     }
 
-    downloadFinished();
+    // do whatever is needed to add a checksum to the http upload request.
+    // in any case, the validator will emit signal startUpload to let the flow
+    // continue in slotStartUpload here.
+    _validator = new TransmissionChecksumValidator( _tmpFile.fileName() );
```

Comment:
This is handled as child of the jobs now, see 5b5a636cc172d3406a06be2da2533dd5df9edc73